Sources/SwiftInspectorVisitors/TypeDescription.swift

```diff
@@ -0,0 +1,238 @@
+// Created by Dan Federman on 2/2/21.
+//
+// Copyright © 2021 Dan Federman
+//
+// Distributed under the MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import SwiftSyntax
+
+public enum TypeDescription: Codable, Equatable, CustomStringConvertible {
+  case simple(name: String)
+  indirect case member(name: String, baseType: TypeDescription)
```

Comment:
Member is a pretty specific concept in Swift. I've heard it refer to instance properties and functions of a type. I think you are using `.member` here when we have a nested type. What about `.nested`?

Reply:
I like diverging a bit from how Swift works internally to make the API easier. Also changing `baseType` to `parentType`.

56de1fa628f88bda61de9bbbb7dd9ed53bd69a81

Reply:
also e4b38ac and c59a340 😅

Reply:
Ah I didn't realize that you were basing "member" on SwiftSyntax types. Yeah I think we can diverge if it's more clear.